src/networking.c

```diff
@@ -1775,8 +1777,17 @@ int freeClientsInAsyncFreeQueue(void) {
     while ((ln = listNext(&li)) != NULL) {
         client *c = listNodeValue(ln);
 
-        if (c->flags & CLIENT_PROTECTED) continue;
-
+        if (c->flags & CLIENT_PROTECTED) {
+            if (!(c->flags & CLIENT_REPL_RDB_CHANNEL)) continue;
```

Comment:
i think this patter would be nicer
```c
        if (c->flags & CLIENT_PROTECTED && (c->flags & CLIENT_REPL_RDB_CHANNEL)) {
...
        }
        if (c->flags & CLIENT_PROTECTED) continue;
```
this way that whole block is just about RDB CHANNEL

to be honest, i'm not sure we should re-use the PROTECTED flag, maybe we should rely on a new flag, or maybe the RDB_CHANNEL one is sufficient.
it's true that this flag does almost exactly what we need, but on the other hand, other users of this flag set it together with removing read and write handlers, and we're not, so it could cause some confusion or issues some day.

i don't mind keeping what you did, just saying i feel a little bit uncomfortable

Reply:
The RDB flag is sufficient however we use it for logs and other stuff so turning off the flag will cause confusion. I agree that using `CLIENT_PROTECTED` may cause wired bugs in the future. I will use dedicated flag.